libs/datasets/timeseries.py

```diff
@@ -202,6 +203,17 @@ def from_source(
         state_fips = data.loc[is_state, CommonFields.STATE].map(us_state_abbrev.ABBREV_US_FIPS)
         data.loc[is_state, CommonFields.FIPS] = state_fips
 
+        no_fips = data[CommonFields.FIPS].isnull()
+        if no_fips.any():
+            print(f"Dropping rows without FIPS in {source}:\n{data.loc[no_fips]}")
+            data = data.loc[~no_fips]
+
+        if data.duplicated(COMMON_FIELDS_TIMESERIES_KEYS, keep=False).any():
+            raise ValueError(
+                f"Duplicates in {source}:\n"
+                f"{data.loc[data.duplicated(COMMON_FIELDS_TIMESERIES_KEYS)]}"
```

Comment:
If possible, I'd prefer not to include the data in the Exception message.  This makes it difficult to combine common alerts and will (I believe) create individual errors in sentry.  I've gotten around this by either setting it as a variable before the exception or creating a new exception with this as an input, something like:
```python
class DuplicateRowError(Exception):
    def __init__(self, data_source, duplicated_data):
        super().__init__("Duplicates in Data Source")
        self.data_source = data_source
        self.duplicated_data = duplicated_data

```

Reply:
I created a new exception... though this is not expected to happen regularly because the build will stop. FYI: In the past I've used attrs to make exception subclasses but can't do that with pydantic: https://stefan.sofa-rockers.org/2020/05/29/attrs-dataclasses-pydantic/